crates/core/src/db/queries/metadatas.rs

```diff
@@ -145,6 +143,7 @@ pub fn list(
             metadata_jsons::image,
         ))
         .distinct()
+        .order(metadatas::address.asc())
```

Comment:
This got us into trouble previously with query times.  Should make sure this doesn't result in any query timeouts first.

Reply:
If the user hasn't passed an order, we shouldn't order.

Reply:
I did. We have to have order for limits.

Reply:
Very skeptical of ordering this.  Will check the query plan but I already had to remove an `ORDER BY` for this query because of the timeouts.

Reply:
@kespinola we indexed name and the resulting index scan still caused the query to take 30+ minutes uncached

Reply:
I'll get some query plans posted.

Reply:
With order by
```
EXPLAIN ANALYZE SELECT DISTINCT metadatas.address, metadatas.name, metadatas.seller_fee_basis_points, metadatas.mint_address, metadatas.primary_sale_happened, metadata_jsons.description, metadata_jsons.image FROM ((metadatas INNER JOIN metadata_creators ON metadatas.address = metadata_creators.metadata_address) INNER JOIN metadata_jsons ON metadatas.address = metadata_jsons.metadata_address) WHERE metadata_creators.creator_address IN ('5ujZ8BBFXpcByNbQdxANxbydxUhz5hVavpapvrEUc2Da', 'HMp36xN691BjS2jDyTLtviuG5KrXFzjGzjgS8UJytZh8') AND metadata_creators.verified = true ORDER BY metadatas.address ASC LIMIT 25 OFFSET 0
;
                                                                                                  QUERY PLAN
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Limit  (cost=9436.48..9439.24 rows=25 width=301) (actual time=144.069..179.325 rows=25 loops=1)
   ->  Unique  (cost=9436.48..9598.28 rows=1467 width=301) (actual time=144.068..179.320 rows=25 loops=1)
         ->  Gather Merge  (cost=9436.48..9593.15 rows=1467 width=301) (actual time=144.067..179.307 rows=25 loops=1)
               Workers Planned: 1
               Workers Launched: 1
               ->  Sort  (cost=8436.48..8436.91 rows=863 width=301) (actual time=139.840..139.847 rows=83 loops=2)
                     Sort Key: metadatas.address, metadatas.name, metadatas.seller_fee_basis_points, metadatas.mint_address, metadatas.primary_sale_happened, metadata_jsons.description, metadata_jsons.image
                     Sort Method: quicksort  Memory: 3174kB
                     Worker 0:  Sort Method: quicksort  Memory: 1909kB
                     ->  Nested Loop  (cost=7.34..8428.06 rows=863 width=301) (actual time=1.924..131.842 rows=4650 loops=2)
                           Join Filter: ((metadata_creators.metadata_address)::text = (metadatas.address)::text)
                           ->  Nested Loop  (cost=7.22..8114.47 rows=1096 width=280) (actual time=1.871..79.182 rows=4650 loops=2)
                                 ->  Parallel Bitmap Heap Scan on metadata_creators  (cost=7.11..3626.86 rows=1096 width=44) (actual time=1.816..20.352 rows=4650 loops=2)
                                       Recheck Cond: (((creator_address)::text = ANY ('{5ujZ8BBFXpcByNbQdxANxbydxUhz5hVavpapvrEUc2Da,HMp36xN691BjS2jDyTLtviuG5KrXFzjGzjgS8UJytZh8}'::text[])) AND verified)
                                       Heap Blocks: exact=5733
                                       ->  Bitmap Index Scan on mcvrfd_idx  (cost=0.00..7.02 rows=1864 width=0) (actual time=2.128..2.129 rows=9424 loops=1)
                                             Index Cond: ((creator_address)::text = ANY ('{5ujZ8BBFXpcByNbQdxANxbydxUhz5hVavpapvrEUc2Da,HMp36xN691BjS2jDyTLtviuG5KrXFzjGzjgS8UJytZh8}'::text[]))
                                 ->  Index Scan using metadata_jsons_metadata_address_index on metadata_jsons  (cost=0.11..4.09 rows=1 width=236) (actual time=0.012..0.012 rows=1 loops=9299)
                                       Index Cond: ((metadata_address)::text = (metadata_creators.metadata_address)::text)
                           ->  Index Scan using metadatas_pkey on metadatas  (cost=0.11..0.28 rows=1 width=109) (actual time=0.011..0.011 rows=1 loops=9299)
                                 Index Cond: ((address)::text = (metadata_jsons.metadata_address)::text)
 Planning Time: 0.719 ms
 Execution Time: 179.616 ms
(23 rows)
```

Reply:
Are these the fast-track for just creator filters or the full query?

Reply:
Here is example that doesn't use the fast track though its actually faster. I do see seq scan on listings so need to add index there.
```
EXPLAIN ANALYZE SELECT DISTINCT metadatas.address, metadatas.name, metadatas.seller_fee_basis_points, metadatas.mint_address, metadatas.primary_sale_happened, metadata_jsons.description, metadata_jsons.image FROM (((((metadatas INNER JOIN metadata_creators ON metadatas.address = metadata_creators.metadata_address) INNER JOIN metadata_jsons ON metadatas.address = metadata_jsons.metadata_address) INNER JOIN token_accounts ON metadatas.mint_address = token_accounts.mint_address) LEFT OUTER JOIN listing_receipts ON metadatas.address = listing_receipts.metadata) LEFT OUTER JOIN attributes ON metadatas.address = attributes.metadata_address) WHERE metadata_creators.creator_address IN ('5ujZ8BBFXpcByNbQdxANxbydxUhz5hVavpapvrEUc2Da', 'HMp36xN691BjS2jDyTLtviuG5KrXFzjGzjgS8UJytZh8') AND metadata_creators.verified = true AND token_accounts.amount = 1 AND token_accounts.owner_address IN ('CqC5dNspaUVbUz6QDbNcodpTfgbw1nZ4ra2nectDv1ak') ORDER BY metadatas.name DESC LIMIT 25 OFFSET 0;
                                                                                                  QUERY PLAN
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Limit  (cost=873.50..873.52 rows=5 width=301) (actual time=0.653..0.663 rows=1 loops=1)
   ->  Unique  (cost=873.50..873.52 rows=5 width=301) (actual time=0.652..0.661 rows=1 loops=1)
         ->  Sort  (cost=873.50..873.50 rows=5 width=301) (actual time=0.651..0.654 rows=8 loops=1)
               Sort Key: metadatas.name DESC, metadatas.address, metadatas.seller_fee_basis_points, metadatas.mint_address, metadatas.primary_sale_happened, metadata_jsons.description, metadata_jsons.image
               Sort Method: quicksort  Memory: 29kB
               ->  Nested Loop Left Join  (cost=0.45..873.49 rows=5 width=301) (actual time=0.317..0.632 rows=8 loops=1)
                     ->  Nested Loop Left Join  (cost=0.34..871.35 rows=1 width=301) (actual time=0.302..0.612 rows=1 loops=1)
                           Join Filter: ((metadatas.address)::text = (listing_receipts.metadata)::text)
                           Rows Removed by Join Filter: 24
                           ->  Nested Loop  (cost=0.34..869.19 rows=1 width=301) (actual time=0.287..0.597 rows=1 loops=1)
                                 ->  Nested Loop  (cost=0.23..868.75 rows=1 width=153) (actual time=0.275..0.584 rows=1 loops=1)
                                       ->  Nested Loop  (cost=0.11..831.84 rows=104 width=109) (actual time=0.041..0.297 rows=20 loops=1)
                                             ->  Index Scan using tok_own_idx on token_accounts  (cost=0.11..415.06 rows=104 width=44) (actual time=0.032..0.197 rows=22 loops=1)
                                                   Index Cond: ((owner_address)::text = 'CqC5dNspaUVbUz6QDbNcodpTfgbw1nZ4ra2nectDv1ak'::text)
                                                   Filter: (amount = 1)
                                                   Rows Removed by Filter: 61
                                             ->  Index Scan using metadatas_mint_address_index on metadatas  (cost=0.00..4.00 rows=1 width=109) (actual time=0.004..0.004 rows=1 loops=22)
                                                   Index Cond: ((mint_address)::text = (token_accounts.mint_address)::text)
                                       ->  Index Scan using metadata_creators_metadata_address_idx on metadata_creators  (cost=0.11..0.35 rows=1 width=44) (actual time=0.014..0.014 rows=0 loops=20)
                                             Index Cond: ((metadata_address)::text = (metadatas.address)::text)
                                             Filter: (verified AND ((creator_address)::text = ANY ('{5ujZ8BBFXpcByNbQdxANxbydxUhz5hVavpapvrEUc2Da,HMp36xN691BjS2jDyTLtviuG5KrXFzjGzjgS8UJytZh8}'::text[])))
                                             Rows Removed by Filter: 2
                                 ->  Index Scan using metadata_jsons_metadata_address_index on metadata_jsons  (cost=0.11..0.44 rows=1 width=236) (actual time=0.011..0.011 rows=1 loops=1)
                                       Index Cond: ((metadata_address)::text = (metadatas.address)::text)
                           ->  Seq Scan on listing_receipts  (cost=0.00..2.07 rows=24 width=45) (actual time=0.006..0.009 rows=24 loops=1)
                     ->  Index Only Scan using maddy_attrs on attributes  (cost=0.11..1.40 rows=248 width=44) (actual time=0.014..0.018 rows=8 loops=1)
                           Index Cond: (metadata_address = (metadatas.address)::text)
                           Heap Fetches: 8
 Planning Time: 3.800 ms
 Execution Time: 0.738 ms
(30 rows)
```